pkg/server/handler/clustermanager/client.go

```diff
@@ -56,8 +55,12 @@ func NewClient(client client.Client) *Client {
 	}
 }
 
+type appBundleLister interface {
```

Comment:
#Is there a reason for this rather than just passing around the concrete type?  I don't see any need for the abstraction besides limiting what a user can do, which is fine I guess.

Even then is there a reason we cannot just define the interface in `internal/applicationbundle/interfaces.go`?  Seems like it should live with the client itself in a shared library, for sharing 🤗